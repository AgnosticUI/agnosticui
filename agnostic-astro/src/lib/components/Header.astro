---
import XElement from 'astro-xelement'
import { pluckClassesFromAttrs } from '../../shared/pluckClassesFromAttrs';

import 'agnostic-css/src/components/header/header.css';

/**
 * This is our main <header><div class="header-content>...</div></header> header structure.
 * The should should be an unordered list using the following components:
 * ul: AgHeaderNav
 *    li: AgHeaderNavItem, 
 *    li: AgHeaderNavItem, 
 * and so on...
 */
const { div: AgHeaderContent, header: AgHeader } = XElement;
const {...attrs} = Astro.props
const { additionalClassList } = pluckClassesFromAttrs(attrs);
---
<AgHeader class:list={['header', {...additionalClassList}]}>
  <AgHeaderContent class='header-content'>
    <slot />
  </AgHeaderContent>
</AgHeader>
