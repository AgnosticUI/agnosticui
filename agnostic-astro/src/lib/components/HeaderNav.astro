---
import XElement from 'astro-xelement'
import { pluckClassesFromAttrs } from '../../shared/pluckClassesFromAttrs';
import 'agnostic-css/src/components/header/headernav.css';

/**
 * This represents the <nav><ul><slot /></ul></nav>; and the slot needs to be
 * a list of HeaderNavItem's (which are <li>'s under-the hood)
 */
const { ul: AgHeaderNav, nav: AgNav} = XElement;

const {...attrs} = Astro.props

const { additionalClassList } = pluckClassesFromAttrs(attrs);
---
<AgNav>
  <AgHeaderNav class:list={[
    "header-nav",
    {...additionalClassList}
  ]}
  >
    <slot />
  </AgHeaderNav>
</AgNav>
