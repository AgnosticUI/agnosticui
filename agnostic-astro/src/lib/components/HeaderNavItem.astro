---
import XElement from 'astro-xelement'
import { pluckClassesFromAttrs } from '../../shared/pluckClassesFromAttrs';
import 'agnostic-css/src/components/header/headernavitem.css';

/**
 * This represents the list of <li>'s that go within the <nav><ul><slot /></ul></nav>
 * structure of HeaderNav.astro. Consumer should project whatever they want within this
 * component's slot. But, ideally, it will be a link and a visually hidden label to allow
 * assistive devices to announce:
 * SLOT example:
 *  <span class="screenreader-only">View our products</span>
 *  <a href="#products">{{ products }}</a>
 * And the above slot will get wrapped inthe <li>...</li> we're producing here.
 */
const { li: AgHeaderNavItem } = XElement;

const {...attrs} = Astro.props

const { additionalClassList } = pluckClassesFromAttrs(attrs);
---
  <AgHeaderNavItem class:list={[
    "header-nav-item",
    {...additionalClassList}
  ]}
  >
    <slot />
  </AgHeaderNavItem>
