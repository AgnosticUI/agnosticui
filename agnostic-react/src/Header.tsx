// import PropTypes from 'prop-types';
import { FC, ReactNode } from 'react';
import styles from './header.module.css';

export interface HeaderNavProps {
  css?: string;
  children: ReactNode;
}

export const HeaderNav: FC<HeaderNavProps> = ({ css = '', children }) => {
  const klasses = [styles.headerNav, css ? `${css}` : ''];
  const classes = klasses.filter((klass) => klass.length).join(' ');
  return (
    <nav>
      <ul className={classes}>{children}</ul>
    </nav>
  );
};

export interface HeaderNavItemProps {
  css?: string;
  children: ReactNode;
}
export const HeaderNavItem: FC<HeaderNavItemProps> = ({ css = '', children }) => {
  const klasses = [styles.headerNavItem, css ? `${css}` : ''];
  const classes = klasses.filter((klass) => klass.length).join(' ');
  return <li className={classes}>{children}</li>;
};

export interface HeaderProps {
  isSticky?: boolean;
  isSkinned?: boolean;
  css?: string;
  children: ReactNode;
}

export const Header: FC<HeaderProps> = ({
  css = '',
  isSticky = false,
  isSkinned = true,
  children,
}) => {
  const klasses = [
    isSkinned ? styles.header : styles.headerBase,
    isSticky ? styles.sticky : '',
    css ? `${css}` : '',
  ];
  const classes = klasses.filter((klass) => klass.length).join(' ');

  return (
    <header className={classes}>
      <div className={styles.headerContent}>{children}</div>
    </header>
  );
};
