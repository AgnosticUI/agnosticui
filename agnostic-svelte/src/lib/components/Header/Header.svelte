<style>
  .header,
  .header-base {
    display: block;
  }

  .header-base :global(img),
  .header :global(img) {
    max-width: 100%;
    height: auto;
  }

  .header,
  .header-skin {
    background-color: var(
      --agnostic-header-background-color,
      var(--agnostic-light)
    );
    box-shadow: var(--agnostic-header-box-shadow-hor, 0)
      var(--agnostic-header-box-shadow-ver, 1px)
      var(--agnostic-header-box-shadow-blur, 5px)
      var(--agnostic-header-box-shadow-spread, 2px)
      var(--agnostic-header-box-shadow-color, rgb(0 0 0 / 10%));
    font-family: var(
      --agnostic-header-font-family,
      var(--agnostic-font-family-body)
    );
    border-bottom: 1px solid
      var(--agnostic-header-border-color, var(--agnostic-gray-light));
    padding-block-start: var(--agnostic-vertical-pad, 0.5rem);
    padding-block-end: var(--agnostic-vertical-pad, 0.5rem);
    padding-inline-start: var(--fluid-24);
    padding-inline-end: var(--fluid-24);
  }

  .header-content {
    width: var(--agnostic-header-content-width, 960px);
    max-width: 100%;
    margin: 0 auto;
    display: flex;
    justify-content: space-around;
    align-items: center;
    flex-flow: wrap column;
  }

  /**
 * If you make your header sticky, you should likely consider that jump links
 * e.g. <h1 id="Jump-Here">... will tuck underneath the header! Here's a recommendation
 * for how to circumnavigate this issue via scroll-margin-top:
  h1[id],
  h2[id],
  h3[id],
  h4[id],
  h5[id],
  h6[id] {
    scroll-margin-top: 80px;
  }
 * In this case I have a 64px tall header so I've added 16px for clearance -- you'll need
 * to curate your own use case, but scroll-margin-top is super useful for this use case.
 */

  .header-sticky {
    position: relative;
    top: 0;
    z-index: 10;
  }

  @media (min-width: 960px) {
    .header-sticky {
      position: sticky;
    }

    .header-content {
      flex-direction: row;
      justify-content: space-between;
    }

    .header-content-start {
      justify-content: flex-start;
    }

    .header-content-end {
      justify-content: flex-end;
    }
  }
</style>

<script>
  export let isSticky = false;
  export let isSkinned = true;
  export let isHeaderContentStart = false;
  export let isHeaderContentEnd = false;
  export let css = "";

  const klasses = [
    isSkinned ? "header" : "header-base",
    isSticky ? "header-sticky" : "",
    css ? `${css}` : "",
  ]
    .filter((cl) => cl.length)
    .join(" ");

  const headerContentClasses = [
    "header-content",
    isHeaderContentStart ? "header-content-start" : "",
    isHeaderContentEnd ? "header-content-end" : "",
  ]
    .filter((cl) => cl.length)
    .join(" ");
</script>

<div class="{klasses}">
  <div class="{headerContentClasses}">
    <slot name="logoleft" />
    <slot />
    <slot name="logoright" />
  </div>
</div>
