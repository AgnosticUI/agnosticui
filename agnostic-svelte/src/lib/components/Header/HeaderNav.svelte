<style>
/* Goes on the <ul> */
.header-nav {
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
}

@media (min-width: 960px) {
  .header-nav {
    flex-direction: row;
  }
}

</style>

<script>
  export let css = "";
  const containerClasses = [css ? `${css}` : ""].filter((c) => c.length);
</script>

<nav class={containerClasses}>
  <ul class="header-nav">
    <slot />
  </ul>
</nav>
