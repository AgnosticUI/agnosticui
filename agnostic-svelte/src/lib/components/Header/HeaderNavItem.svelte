<style>
.header-nav-item {
  display: inline-block;
}

.header-nav-item:not(:last-child) {
  margin-inline-end: initial;
  margin-block-end: var(--fluid-8);
}

.header-nav-item :global(a) {
  color: var(--agnostic-header-color, var(--agnostic-font-color));
}

@media (min-width: 960px) {
  .header-nav-item:not(:last-child) {
    margin-inline-end: var(--agnostic-header-nav-spacing, var(--fluid-32));
    margin-block-end: initial;
  }
}

</style>

<script>
  export let css = "";
  let klasses = ["header-nav-item", css ? `${css}` : ""];
  klasses = klasses.filter((klass) => klass.length);
  klasses = klasses.join(" ");
</script>

<li class="{klasses}">
  <slot />
</li>
