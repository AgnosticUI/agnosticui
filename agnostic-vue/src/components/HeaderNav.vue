<style module>
/* Goes on the <ul> */
.header-nav {
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
}

@media (min-width: 960px) {
  .header-nav {
    flex-direction: row;
  }
}

</style>
<template>
  <nav>
    <ul :class="classes">
      <slot />
    </ul>
  </nav>
</template>
<script>
export default {
  name: "AgHeaderNav",
  props: {
    css: {
      type: String,
      default: "",
    },
  },
  computed: {
    classes() {
      return {
        [this.$style["header-nav"]]: true,
        [`${this.css}`]: !!this.css,
      };
    },
  },
};
</script>
