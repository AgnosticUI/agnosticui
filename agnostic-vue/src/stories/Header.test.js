import { render } from '@testing-library/vue';
import Header from './Header.vue';
import HeaderNav from './HeaderNav.vue';
import HeaderNavItem from './HeaderNavItem.vue';

//   template: '<agnostic-header v-bind="$props"><template v-slot:logoleft><a href="#">Logo on Left</a></template><template v-slot:headernav><agnostic-header-nav><ul><li><a href="#">Home</a></li><li><a href="#">About</a></li></ul></agnostic-header-nav></template><template v-slot:logoright><a href="#">Logo on Right</a></template></agnostic-header>'
test('Header renders', async () => {
  const { container, getByText } = render(Header, {
    components: {
      HeaderNav,
      HeaderNavItem,
    },
    slots: {
      logoleft: '<a href="#">Logo Left</a>',
      logoright: '<a href="#">Logo Right</a>',
      headernav: '<HeaderNav><HeaderNavItem><a href="#">Home</a></HeaderNavItem><HeaderNavItem><a href="#">About</a></HeaderNavItem></HeaderNav>',
    },
  });
  getByText('Home');
  getByText('About');
  getByText('Logo Left');
  getByText('Logo Right');
  expect(container).toMatchSnapshot();
});
