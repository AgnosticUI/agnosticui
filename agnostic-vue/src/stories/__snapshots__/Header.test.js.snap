// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="header-content"
    >
      <a
        href="#"
      >
        Logo Left
      </a>
       
      <nav>
        <ul
          class="header-nav"
        >
          <li
            class="header-nav-item"
          >
            <a
              href="#"
            >
              Home
            </a>
          </li>
          <li
            class="header-nav-item"
          >
            <a
              href="#"
            >
              About
            </a>
          </li>
        </ul>
      </nav>
       
      <a
        href="#"
      >
        Logo Right
      </a>
    </div>
  </div>
</div>
`;
